### ISSUE TITLE
RUF031: False positive when using with unpacking 

### ISSUE THREAD

**Issue by alexfikl** [astral-sh/ruff#12776]
The following run should not indicate any errors
```
>> ruff check --isolated --preview --select RUF031 --target-version py38 subscript-tuple.py
subscript-tuple.py:1:3: RUF031 [*] Avoid parentheses for tuples in subscripts.
|
1 | d[(*indices, 5)]
|   ^^^^^^^^^^^^^ RUF031
|
= help: Remove the parentheses.

Found 1 error.
[*] 1 fixable with the `--fix` option.
```
As far as I know, `d[*indices, 5]` is invalid syntax on Python 3.8, but should work for Python >=3.9 (with the new PEG parser, I think? definitely works on 3.12).

Expected result:
* No errors / fixes reported when setting `--target-version py38`.
* The recommended fix when setting `--target-version py39` and above.

xref: CI failure resulting from the fix: https://github.com/inducer/pymbolic/actions/runs/10315281929/job/28555159989?pr=144

**Comment by MichaReiser** [comment]
Thanks for reporting this preview error!

@dylwil3  do you want to take this. Surprising how many edge cases there are to this rule :open_mouth:

**Comment by dylwil3** [comment]
Sure!

This has been a good learning experience in both Rust and the intricacies of Python, so I'm happy to keep taking on RUF031 bugs 😄

**Comment by dylwil3** [comment]
The CI link above seems to also show a syntax error in Python 3.9, unless I'm reading it wrong:

https://github.com/inducer/pymbolic/actions/runs/10315281929/job/28555160514?pr=144

I can confirm 3.11 is ok, but I gotta go check 3.10. Will update PR once I get a chance to do that.

**Comment by alexfikl** [comment]
> The CI link above seems to also show a syntax error in Python 3.9, unless I'm reading it wrong:

Ah, yes! I didn't check that because it got cancelled, but it seems like 3.9 also doesn't support unpacking in the subscript. Sorry for the confusion.

**Comment by dylwil3** [comment]
Looks like this is a syntax error in 3.10 as well. It was allowed starting in 3.11, hidden in [PEP-646](https://peps.python.org/pep-0646/#change-1-star-expressions-in-indexes)
